src/node_zlib.cc

```diff
@@ -476,9 +468,14 @@ class ZCtx : public AsyncWrap {
       memcpy(dictionary, dictionary_, dictionary_len);
     }
 
-    Init(ctx, level, windowBits, memLevel, strategy, write_result,
-         write_js_callback, dictionary, dictionary_len);
+    bool ret = Init(ctx, level, windowBits, memLevel, strategy, write_result,
+                    write_js_callback, dictionary, dictionary_len);
+    if (!ret) goto end;
```

Comment:
Any reason we need a goto here? (other than one less line of duplicate code?)

Reply:
just that... didn't want to duplicate the line of code